gammapy/cube/make.py

```diff
@@ -257,3 +297,359 @@ def _check_selection(selection):
+        maplist: the list of dictionaries returned by MapMakerRing.run()
+                or MapMakerRing.run_images()
+        convolution_radius: `~astropy.units.Quantity`
+            The convolution radius for Tophat2DKernel
+
+            If not specified, returns the uncorrelated significance map
+
+        Returns
+        -------
+        significance_map_off: `~gammapy.maps.Map`
+        Stacked significance map for the off regions region.
+
+        """
+
+        significance_map = self.significance_map(maplist, convolution_radius)
+        if significance_map.geom.is_image:
+            return significance_map * self.exclusion_mask.get_image_by_idx([0])
+        else:
+            return significance_map * self.exclusion_mask
+
+    def excess_map(self, maplist):
+        """
+        excess = on - alpha * off
+
+        Parameters
+        ---------
+        maplist: the list of dictionaries returned by MapMakerRing.run()
+                or MapMakerRing.run_images()
+
+        Returns
+        -------
+        excess_map: `~gammapy.maps.Map`
+        Stacked excess map for the entire region for each pixel.
+        """
+
+        selection = ["on", "alpha", "off"]
+        stacked_map = {}
+        for name in selection:
+            stacked_map[name] = Map.from_geom(maplist[0]["counts"].geom)
+        for amap in maplist:
+            stacked_map["on"] += amap["counts"]
+            stacked_map["off"] += amap["off"]
+            stacked_map["alpha"] += amap["off"] * amap["alpha"]
+        stacked_map["alpha"] /= stacked_map["off"]
+        excess_map = stacked_map["on"] - stacked_map["alpha"] * stacked_map["off"]
+
+        return excess_map
+
+    def excess_map_off(self, maplist):
+        """
+        returns the excess map with exclusion region applied
+
+        Parameters
+        ---------
+        maplist: the list of dictionaries returned by MapMakerRing.run()
+                or MapMakerRing.run_images()
+
+        Returns
+        -------
+        excess_map_off: `~gammapy.maps.Map`
+        Stacked excess map for the off for each pixel.
+        """
+
+        excess_map = self.excess_map(maplist)
+        if excess_map.geom.is_image:
+            return excess_map * self.exclusion_mask.get_image_by_idx([0])
+        else:
+            return excess_map * self.exclusion_mask
+
+    def plot(self, maplist, convolution_radius=None, idx=[0]):
+        """Makes some useful plots: the significance and excess maps,
+        and the histograms of the significance.
+
+        This can be done for each slice of the map by specifying idx.
+        In case the maps are aleady in 2D, no idx is necessary
+
+        Parameters
+        ----------
+        maplist : list of dictionaries of maps
+
+        idx : tuple, optional
+            Tuple of scalar indices for each non spatial dimension of the map.
+            Tuple should be ordered as (I_0, ..., I_n).
+
+        """
+        excess_map = self.excess_map(maplist)
```

Comment:
This returns a stacked excess map, but what if the user wants to plot `excess` and `significance` only for a single observation?